### ISSUE TITLE
cannot upload new calendar items to horde

### ISSUE THREAD

**Issue by gfa** [untitaker/vdirsyncer#251]
i've created very simple calendar entry using khal and now i'm trying to sync to my server

```
debug: Old content: u'BEGIN:VCALENDAR\r\nVERSION:2.0\r\nPRODID:-//CALENDARSERVER.ORG//NONSGML Version 1//EN\r\nBEGIN:VTIMEZONE\r\nTZID:Asia/Shanghai\r\nBEGIN:STANDARD\r\nDTSTART;VALUE=DATE-TIME:19910914T230000\r\nTZNAME:CST\r\nTZOFFSETFROM:+0900\r\nTZOFFSETTO:+0800\r\nEND:STANDARD\r\nEND:VTIMEZONE\r\nBEGIN:VEVENT\r\nSUMMARY:test\r\nDTSTART;TZID=Asia/Shanghai;VALUE=DATE-TIME:20150806T170000\r\nDTEND;TZID=Asia/Shanghai;VALUE=DATE-TIME:20150806T180000\r\nDTSTAMP;VALUE=DATE-TIME:20150805T160848Z\r\nUID:CZOGUN2KL8O3QUMLW4K23L57736BDX0TD51R\r\nSEQUENCE:1\r\nDESCRIPTION:\r\nLOCATION:\r\nEND:VEVENT\r\nEND:VCALENDAR\r\n'
debug: New content: u'BEGIN:VCALENDAR\nVERSION:2.0\nX-WR-CALNAME:Calendar of gfa@zumbi.com.ar\nPRODID:-//The Horde Project//Horde iCalendar Library//EN\nBEGIN:VEVENT\nDTSTART;TZID=Asia/Shanghai:20150806T170000\nDTEND;TZID=Asia/Shanghai:20150806T180000\nDTSTAMP:20150805T080937Z\nUID:CZOGUN2KL8O3QUMLW4K23L57736BDX0TD51R\nCREATED:20150805T080902Z\nLAST-MODIFIED:20150805T080937Z\nSUMMARY:test\nCLASS:PUBLIC\nSTATUS:CONFIRMED\nTRANSP:OPAQUE\nEND:VEVENT\nBEGIN:VTIMEZONE\nTZID:Asia/Shanghai\nBEGIN:DAYLIGHT\nTZOFFSETFROM:+0800\nTZOFFSETTO:+0900\nDTSTART:19400603T000000\nTZNAME:CDT\nEND:DAYLIGHT\nBEGIN:STANDARD\nTZOFFSETFROM:+0900\nTZOFFSETTO:+0800\nTZNAME:CST\nEND:STANDARD\nBEGIN:DAYLIGHT\nTZOFFSETFROM:+0800\nTZOFFSETTO:+0900\nDTSTART:19410316T000000\nTZNAME:CDT\nEND:DAYLIGHT\nBEGIN:DAYLIGHT\nTZOFFSETFROM:+0800\nTZOFFSETTO:+0900\nDTSTART:19860504T000000\nTZNAME:CDT\nEND:DAYLIGHT\nBEGIN:STANDARD\nTZOFFSETFROM:+0900\nTZOFFSETTO:+0800\nDTSTART:19860914T000000\nRRULE:FREQ=YEARLY;BYMONTH=9;BYMONTHDAY=11,12,13,14,15,16,17;BYDAY=1SU;UNTIL\n =19910914T160000Z\nTZNAME:CST\nEND:STANDARD\nBEGIN:DAYLIGHT\nTZOFFSETFROM:+0800\nTZOFFSETTO:+0900\nDTSTART:19870412T000000\nRRULE:FREQ=YEARLY;BYMONTH=4;BYMONTHDAY=10,11,12,13,14,15,16;BYDAY=1SU;UNTIL\n =19910413T160000Z\nTZNAME:CDT\nEND:DAYLIGHT\nEND:VTIMEZONE\nEND:VCALENDAR\n'
Unhandled exception occured for calendar.
Traceback (most recent call last):
  File "build/bdist.linux-x86_64/egg/vdirsyncer/cli/tasks.py", line 65, in sync_collection
    force_delete=force_delete
  File "build/bdist.linux-x86_64/egg/vdirsyncer/sync.py", line 175, in sync
    action(a_info, b_info, conflict_resolution)
  File "build/bdist.linux-x86_64/egg/vdirsyncer/sync.py", line 276, in inner
    _action_update(ident, a, b)(a, b, conflict_resolution)
  File "build/bdist.linux-x86_64/egg/vdirsyncer/sync.py", line 220, in inner
    dest_meta['etag'])
  File "build/bdist.linux-x86_64/egg/vdirsyncer/storage/base.py", line 17, in inner
    return f(self, *args, **kwargs)
  File "build/bdist.linux-x86_64/egg/vdirsyncer/storage/base.py", line 17, in inner
    return f(self, *args, **kwargs)
  File "build/bdist.linux-x86_64/egg/vdirsyncer/storage/dav.py", line 494, in update
    href, etag = self._put(self._normalize_href(href), item, etag)
  File "build/bdist.linux-x86_64/egg/vdirsyncer/storage/dav.py", line 486, in _put
    'While requesting the etag for {!r}, '
WrongEtagError: While requesting the etag for {!r}, the item content changed.
```
vdirsyncer, version 0.5.3.dev56+ng9229d08

same problem with python3

i would give to you an account on my install to configure in travis for the CI, do you know if is possible to hidden that information?

**Comment by untitaker** [comment]
I will have to remove that check, I didn't think this through properly.

On 5 August 2015 10:27:05 CEST, gustavo panizzo <notifications@github.com> wrote:
>i've created very simple calendar entry using khal and now i'm trying
>to sync to my server
>
>```
>debug: Old content:
>u'BEGIN:VCALENDAR\r\nVERSION:2.0\r\nPRODID:-//CALENDARSERVER.ORG//NONSGML
>Version
>1//EN\r\nBEGIN:VTIMEZONE\r\nTZID:Asia/Shanghai\r\nBEGIN:STANDARD\r\nDTSTART;VALUE=DATE-TIME:19910914T230000\r\nTZNAME:CST\r\nTZOFFSETFROM:+0900\r\nTZOFFSETTO:+0800\r\nEND:STANDARD\r\nEND:VTIMEZONE\r\nBEGIN:VEVENT\r\nSUMMARY:test\r\nDTSTART;TZID=Asia/Shanghai;VALUE=DATE-TIME:20150806T170000\r\nDTEND;TZID=Asia/Shanghai;VALUE=DATE-TIME:20150806T180000\r\nDTSTAMP;VALUE=DATE-TIME:20150805T160848Z\r\nUID:CZOGUN2KL8O3QUMLW4K23L57736BDX0TD51R\r\nSEQUENCE:1\r\nDESCRIPTION:\r\nLOCATION:\r\nEND:VEVENT\r\nEND:VCALENDAR\r\n'
>debug: New content:
>u'BEGIN:VCALENDAR\nVERSION:2.0\nX-WR-CALNAME:Calendar of
>gfa@zumbi.com.ar\nPRODID:-//The Horde Project//Horde iCalendar
>Library//EN\nBEGIN:VEVENT\nDTSTART;TZID=Asia/Shanghai:20150806T170000\nDTEND;TZID=Asia/Shanghai:20150806T180000\nDTSTAMP:20150805T080937Z\nUID:CZOGUN2KL8O3QUMLW4K23L57736BDX0TD51R\nCREATED:20150805T080902Z\nLAST-MODIFIED:20150805T080937Z\nSUMMARY:test\nCLASS:PUBLIC\nSTATUS:CONFIRMED\nTRANSP:OPAQUE\nEND:VEVENT\nBEGIN:VTIMEZONE\nTZID:Asia/Shanghai\nBEGIN:DAYLIGHT\nTZOFFSETFROM:+0800\nTZOFFSETTO:+0900\nDTSTART:19400603T000000\nTZNAME:CDT\nEND:DAYLIGHT\nBEGIN:STANDARD\nTZOFFSETFROM:+0900\nTZOFFSETTO:+0800\nTZNAME:CST\nEND:STANDARD\nBEGIN:DAYLIGHT\nTZOFFSETFROM:+0800\nTZOFFSETTO:+0900\nDTSTART:19410316T000000\nTZNAME:CDT\nEND:DAYLIGHT\nBEGIN:DAYLIGHT\nTZOFFSETFROM:+0800\nTZOFFSETTO:+0900\nDTSTART:19860504T000000\nTZNAME:CDT\nEND:DAYLIGHT\nBEGIN:STANDARD\nTZOFFSETFROM:+0900\nTZOFFSETTO:+0800\nDTSTART:19860914T000000\nRRULE:FREQ=YEARLY;BYMONTH=9
> ;BYMONTH
>DAY=11,12,13,14,15,16,17;BYDAY=1SU;UNTIL\n
>=19910914T160000Z\nTZNAME:CST\nEND:STANDARD\nBEGIN:DAYLIGHT\nTZOFFSETFROM:+0800\nTZOFFSETTO:+0900\nDTSTART:19870412T000000\nRRULE:FREQ=YEARLY;BYMONTH=4;BYMONTHDAY=10,11,12,13,14,15,16;BYDAY=1SU;UNTIL\n
>=19910413T160000Z\nTZNAME:CDT\nEND:DAYLIGHT\nEND:VTIMEZONE\nEND:VCALENDAR\n'
>Unhandled exception occured for calendar.
>Traceback (most recent call last):
>File "build/bdist.linux-x86_64/egg/vdirsyncer/cli/tasks.py", line 65,
>in sync_collection
>    force_delete=force_delete
>File "build/bdist.linux-x86_64/egg/vdirsyncer/sync.py", line 175, in
>sync
>    action(a_info, b_info, conflict_resolution)
>File "build/bdist.linux-x86_64/egg/vdirsyncer/sync.py", line 276, in
>inner
>    _action_update(ident, a, b)(a, b, conflict_resolution)
>File "build/bdist.linux-x86_64/egg/vdirsyncer/sync.py", line 220, in
>inner
>    dest_meta['etag'])
>File "build/bdist.linux-x86_64/egg/vdirsyncer/storage/base.py", line
>17, in inner
>    return f(self, *args, **kwargs)
>File "build/bdist.linux-x86_64/egg/vdirsyncer/storage/base.py", line
>17, in inner
>    return f(self, *args, **kwargs)
>File "build/bdist.linux-x86_64/egg/vdirsyncer/storage/dav.py", line
>494, in update
>    href, etag = self._put(self._normalize_href(href), item, etag)
>File "build/bdist.linux-x86_64/egg/vdirsyncer/storage/dav.py", line
>486, in _put
>    'While requesting the etag for {!r}, '
>WrongEtagError: While requesting the etag for {!r}, the item content
>changed.
>```
>vdirsyncer, version 0.5.3.dev56+ng9229d08
>
>same problem with python3
>
>i would give to you an account on my install to configure in travis for
>the CI, do you know if is possible to hidden that information?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/untitaker/vdirsyncer/issues/251

-- 
Sent from my phone. Please excuse my brevity.

**Comment by untitaker** [comment]
@gfa This should be fixed in master, available with 0.6.0.

**Comment by gfa** [comment]
On Wed, Aug 05, 2015 at 02:45:14PM -0700, Markus Unterwaditzer wrote:
> @gfa This should be fixed in master, available with 0.6.0.
it does, thanks
